### PR TITLE
Created a leetcode leaderboard

### PR DESCRIPTION
**Pull request by elimelt**
simple endpoint for leetcode leaderboard

ex:

```
/leaderboard/leetcode/
/leaderboard/leetcode/?order_by=hard
/leaderboard/leetcode/?order_by=medium
/leaderboard/leetcode/?order_by=recent
/leaderboard/leetcode/?order_by=completion
/leaderboard/leetcode/?updated_within=24
/leaderboard/leetcode/?order_by=hard&updated_within=48
```

### CODE REVIEW

File: server/leaderboard/views.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    serializer_class = LeetcodeStatsSerializer
+    permission_classes = [IsAuthenticatedOrReadOnly]
+
+    def get_queryset(self):
+        latest_stat = LeetcodeStats.objects.order_by("-last_updated").first()
+        if not latest_stat or timezone.now() - latest_stat.last_updated > timedelta(
+            hours=1
+        ):
+            logger.info("Updating Leetcode stats")
+            call_command("update_leetcode_stats", timeout=2)
```
Comment (ShawnCollinge) [inline server/leaderboard/views.py:25]: Does it wait for this? will it be very slow if we have a lot of users?

**Review by ShawnCollinge — APPROVED** [review]
LGTM but please respond to my comment

File: server/leaderboard/views.py (hunk 1 above)
Comment (ShawnCollinge) [reply server/leaderboard/views.py:25]: If it doesn't wait for it, will we have very outdated info then?

Comment (elimelt) [reply server/leaderboard/views.py:25]: yep, could set up cron but I was lazy and didn't want to make deployment more complicated

Comment (elimelt) [reply server/leaderboard/views.py:25]: I will add TODO though

**Pull request merged**